cockatrice/src/releasechannel.cpp

```diff
@@ -0,0 +1,307 @@
+#include "releasechannel.h"
+#include "qt-json/json.h"
+#include "version_string.h"
+
+#include <QNetworkReply>
+#include <QNetworkAccessManager>
+#include <QMessageBox>
+
+#define STABLERELEASE_URL "https://api.github.com/repos/Cockatrice/Cockatrice/releases/latest"
+#define STABLETAG_URL "https://api.github.com/repos/Cockatrice/Cockatrice/git/refs/tags/"
+#define STABLEFILES_URL "https://api.bintray.com/packages/cockatrice/Cockatrice/Cockatrice/files"
+#define STABLEDOWNLOAD_URL "https://dl.bintray.com/cockatrice/Cockatrice/"
+#define STABLEMANUALDOWNLOAD_URL "https://bintray.com/cockatrice/Cockatrice/Cockatrice/_latestVersion#files"
+
+#define DEVRELEASE_URL "https://api.github.com/repos/Cockatrice/Cockatrice/commits/master"
+#define DEVFILES_URL "https://api.bintray.com/packages/cockatrice/Cockatrice/Cockatrice-git/files"
+#define DEVDOWNLOAD_URL "https://dl.bintray.com/cockatrice/Cockatrice/"
+#define DEVMANUALDOWNLOAD_URL "https://bintray.com/cockatrice/Cockatrice/Cockatrice-git/_latestVersion#files"
+#define GIT_SHORT_HASH_LEN 7
+
+ReleaseChannel::ReleaseChannel(int _index, QString _name)
```

Comment:
Seems like the last 2 fields may want to be an UpdateResponse type object instead of the checker class itself.